### ISSUE TITLE
Dashboard reports path paramaters as optional when not

### ISSUE THREAD

**Issue by tomchiverton** [atuttle/Taffy#308]
This resource cfc shows up in the dashboard docs side panel with each parameter labelled "optional string", but when you press "send" the os and id boxes are outlined in red as required.
```
component extends="taffy.core.resource" taffy_uri="/connect/{os}/{id}" {
    function get(string os, string id, string jsonData,string appId){
```
Taffy 3.1.0-rc1

**Comment by atuttle** [comment]
Well there are two things at play, here.

I'm not saying this isn't a bug, I just want to discuss the conflicting inputs and come to the best solution.

As far as your function signature is concerned, the function arguments are all optional. To make them required the signature would have to be, `function get( required string os, required string id, ... )`. This is why the documentation lists them as optional.

Now, the dashboard is smart enough to know that URL tokens are required for the request to match the current resource. Consider resources with `/widgets` (collection) and `/widgets/{id}` (member). If you were to make a request from the dashboard section for the member but neglect to include an ID parameter, at best you would get back a response from the collection. Since you're obviously expecting a member response (because you're working with the member's section of the dashboard), it enforces that URL tokens are required.

Perhaps, then, the best path forward would be to add additional logic to the docs to make sure we label arguments that match url tokens as required. What do you think?

**Comment by tomchiverton** [comment]
Yes, I think the side bar docs should list them as required, the same way that the 'run' code infers them to be 
-- 
Tom
